### ISSUE TITLE
I am having error during the MAKE

### ISSUE THREAD

**Issue by imaddyx** [joswr1ght/asleap#4]
/asleap/utils.c:182: undefined reference to `encrypt'
collect2: error: ld returned 1 exit status
make: *** [Makefile:34: genkeys] Error 1

NOTE: I am using Ubuntu 20.04

**Comment by joswr1ght** [comment]
I don't get this error on Ubuntu:

```
sec504@slingshot:~$ git clone https://github.com/joswr1ght/asleap.git
Cloning into 'asleap'...
remote: Enumerating objects: 36, done.
remote: Total 36 (delta 0), reused 0 (delta 0), pack-reused 36
Unpacking objects: 100% (36/36), done.
sec504@slingshot:~$ cd asleap/
sec504@slingshot:~/asleap$ make
cc -pipe -Wall -D_LINUX -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I../../.. -D_OPENSSL_MD4 -g3 -ggdb -g   -c -o sha1.o sha1.c
cc -pipe -Wall -D_LINUX -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I../../.. -D_OPENSSL_MD4 -g3 -ggdb -g   -c -o common.o common.c
cc -pipe -Wall -D_LINUX -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I../../.. -D_OPENSSL_MD4 -g3 -ggdb -g   -c -o utils.o utils.c
cc -pipe -Wall -D_LINUX -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I../../.. -D_OPENSSL_MD4 -g3 -ggdb -g asleap.c -o asleap common.o utils.o sha1.o -lpcap -lcrypt -lcrypto
asleap.c: In function ‘main’:
asleap.c:1398:6: warning: variable ‘findleaptimeout’ set but not used [-Wunused-but-set-variable]
  int findleaptimeout = 5;
      ^~~~~~~~~~~~~~~
cc -pipe -Wall -D_LINUX -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I../../.. -D_OPENSSL_MD4 -g3 -ggdb -g md4.c genkeys.c -o genkeys common.o utils.o -lpcap -lcrypt -lcrypto
cc -pipe -Wall -D_LINUX -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I../../.. -D_OPENSSL_MD4 -g3 -ggdb -g   -c -o asleap.o asleap.c
asleap.c: In function ‘main’:
asleap.c:1398:6: warning: variable ‘findleaptimeout’ set but not used [-Wunused-but-set-variable]
  int findleaptimeout = 5;
      ^~~~~~~~~~~~~~~
cc -pipe -Wall -D_LINUX -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I../../.. -D_OPENSSL_MD4 -g3 -ggdb -g   -c -o genkeys.o genkeys.c
sec504@slingshot:~/asleap$ uname -a
Linux slingshot 4.15.0-91-generic #92-Ubuntu SMP Fri Feb 28 11:09:48 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
sec504@slingshot:~/asleap$ cat /etc/issue
Ubuntu 18.04.3 LTS \n \l
```

Can you provide some more information about what you are doing? How did you compile this? What platform are you using?

**Comment by imaddyx** [comment]
@joswr1ght Sir, This is the complete error
```
maddy@maddy-Linux:~/wifi/airgeddon/bin$ git clone https://github.com/joswr1ght/asleap.git
Cloning into 'asleap'...
\remote: Enumerating objects: 36, done.
remote: Total 36 (delta 0), reused 0 (delta 0), pack-reused 36
Unpacking objects: 100% (36/36), 112.62 KiB | 236.00 KiB/s, done.
maddy@maddy-Linux:~/wifi/airgeddon/bin$ \cd asleap/
maddy@maddy-Linux:~/wifi/airgeddon/bin/asleap$ make
cc -pipe -Wall -D_LINUX -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I../../.. -D_OPENSSL_MD4 -g3 -ggdb -g   -c -o sha1.o sha1.c
cc -pipe -Wall -D_LINUX -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I../../.. -D_OPENSSL_MD4 -g3 -ggdb -g   -c -o common.o common.c
cc -pipe -Wall -D_LINUX -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I../../.. -D_OPENSSL_MD4 -g3 -ggdb -g   -c -o utils.o utils.c
utils.c: In function ‘DesEncrypt’:
utils.c:179:2: warning: implicit declaration of function ‘setkey’ [-Wimplicit-function-declaration]
  179 |  setkey((char *)crypt_key);
      |  ^~~~~~
utils.c:182:2: warning: implicit declaration of function ‘encrypt’; did you mean ‘crypt’? [-Wimplicit-function-declaration]
  182 |  encrypt((char *)des_input, 0);
      |  ^~~~~~~
      |  crypt
cc -pipe -Wall -D_LINUX -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I../../.. -D_OPENSSL_MD4 -g3 -ggdb -g asleap.c -o asleap common.o utils.o sha1.o -lpcap -lcrypt -lcrypto
asleap.c: In function ‘main’:
asleap.c:1398:6: warning: variable ‘findleaptimeout’ set but not used [-Wunused-but-set-variable]
 1398 |  int findleaptimeout = 5;
      |      ^~~~~~~~~~~~~~~
/usr/bin/ld: utils.o: in function `DesEncrypt':
/home/maddy/wifi/airgeddon/bin/asleap/utils.c:179: undefined reference to `setkey'
/usr/bin/ld: /home/maddy/wifi/airgeddon/bin/asleap/utils.c:182: undefined reference to `encrypt'
collect2: error: ld returned 1 exit status
make: *** [Makefile:31: asleap] Error 1
maddy@maddy-Linux:~/wifi/airgeddon/bin/asleap$ uname -a
Linux maddy-Linux 5.4.0-33-generic #37-Ubuntu SMP Thu May 21 12:53:59 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
maddy@maddy-Linux:~/wifi/airgeddon/bin/asleap$ cat /etc/issue
Ubuntu 20.04 LTS \n \l)
```

**Comment by joswr1ght** [comment]
Thank you! It appears that Ubuntu 20.04's GLIBC removed the `encrypt()` function. I'll look at replacing it with something else and will return to this ticket. Give me a few days.

**Comment by JLP04** [comment]
I am also having this issue on Raspbian

**Comment by imaddyx** [comment]
@joswr1ght Hello sir, is the issue resolved?

**Comment by gordat** [comment]
Hello,

not yet. Still no update of asleap :(

**Comment by ghost** [comment]
still waiting, tried clang and gcc

`Making package: asleap 2.2-2 (Saturday, 28 November, 2020 01:07:18 AM)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found asleap-2.2.tgz
==> Validating source files with md5sums...
    asleap-2.2.tgz ... Passed
==> Extracting sources...
  -> Extracting asleap-2.2.tgz with bsdtar
==> Removing existing $pkgdir/ directory...
==> Starting build()...
gcc -pipe -Wall -D_LINUX -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I../../.. -D_OPENSSL_MD4 -g3 -ggdb -g asleap.c -o asleap common.o utils.o sha1.o -lpcap -lcrypt -lcrypto
gcc -pipe -Wall -D_LINUX -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I../../.. -D_OPENSSL_MD4 -g3 -ggdb -g md4.c genkeys.c -o genkeys common.o utils.o -lpcap -lcrypt -lcrypto
asleap.c: In function ‘main’:
asleap.c:1398:6: warning: variable ‘findleaptimeout’ set but not used [-Wunused-but-set-variable]
 1398 |  int findleaptimeout = 5;
      |      ^~~~~~~~~~~~~~~
/usr/bin/ld: utils.o: in function `DesEncrypt':
/home/echo26/Downloads/asleap/src/asleap-2.2/utils.c:179: undefined reference to `setkey'
/usr/bin/ld: /home/echo26/Downloads/asleap/src/asleap-2.2/utils.c:182: undefined reference to `encrypt'
collect2: error: ld returned 1 exit status
make: *** [Makefile:34: genkeys] Error 1
make: *** Waiting for unfinished jobs....
/usr/bin/ld: utils.o: in function `DesEncrypt':
/home/echo26/Downloads/asleap/src/asleap-2.2/utils.c:179: undefined reference to `setkey'
/usr/bin/ld: /home/echo26/Downloads/asleap/src/asleap-2.2/utils.c:182: undefined reference to `encrypt'
collect2: error: ld returned 1 exit status
make: *** [Makefile:31: asleap] Error 1
==> ERROR: A failure occurred in build().
    Aborting...
`

**Comment by joswr1ght** [comment]
Fixed with 4576d14f; you will need to install `libxcrypt-dev` on your system to build.
